pkg/kn/commands/service/service_show.go

```diff
@@ -0,0 +1,485 @@
+
+	percent int
+	latest  *bool
+
+	logUrl         string
+	timeoutSeconds *int64
+
+	image       string
+	imageDigest string
+	env         []string
+	port        *int32
+}
+
+// [REMOVE COMMENT WHEN MOVING TO 0.7.0]
+// For transition to v1beta1 this command uses the migration approach as described
+// in https://docs.google.com/presentation/d/1mOhnhy8kA4-K9Necct-NeIwysxze_FUule-8u5ZHmwA/edit#slide=id.p
+// With serving 0.6.0 we are at step #1
+// I.e we first look at new fields of the v1alpha1 API before falling back to the original ones.
+// As this command does not any updates, it's just a matter of fallbacks.
+// [/REMOVE COMMENT WHEN MOVING TO 0.7.0]
+
+// Return a new command for describing a service.
+func NewServiceDescribeCommand(p *commands.KnParams) *cobra.Command {
+	command := &cobra.Command{
+		Use:   "show NAME",
+		Short: "Show details for a given service",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if len(args) < 1 {
+				return errors.New("no service name provided")
+			}
+			if len(args) > 1 {
+				return errors.New("more than one service name provided")
+			}
+			serviceName := args[0]
+
+			namespace, err := p.GetNamespace(cmd)
+			if err != nil {
+				return err
+			}
+
+			client, err := p.NewClient(namespace)
+			if err != nil {
+				return err
+			}
+
+			printDetails, err = cmd.Flags().GetBool("details")
+			if err != nil {
+				return err
+			}
+
+			service, err := client.GetService(serviceName)
+			if err != nil {
+				return err
+			}
+			revisionDescs, err := getRevisionDescriptions(client, service, printDetails)
+
+			return describe(cmd.OutOrStdout(), service, revisionDescs)
+		},
+	}
+	flags := command.Flags()
+	commands.AddNamespaceFlags(flags, false)
+	flags.BoolP("details", "d", false, "Show all details.")
+	return command
+}
+
+// Main action describing the service
+func describe(w io.Writer, service *v1alpha1.Service, revisions []*revisionDesc) error {
+	dw := printers.NewPrefixWriter(w)
+
+	// Service info
+	writeService(dw, service)
+	dw.WriteLine()
+	if err := dw.Flush(); err != nil {
+		return err
+	}
+
+	// Revisions summary info
+	writeRevisions(dw, revisions)
+	dw.WriteLine()
+	if err := dw.Flush(); err != nil {
+		return err
+	}
+
+	// Condition info
+	writeConditions(dw, service)
+	if err := dw.Flush(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Write out main service information. Use colors for major items.
+func writeService(dw printers.PrefixWriter, service *v1alpha1.Service) {
+	dw.WriteColsLn(printers.LEVEL_0, l("Name"), service.Name)
+	dw.WriteColsLn(printers.LEVEL_0, l("Namespace"), service.Namespace)
+	dw.WriteColsLn(printers.LEVEL_0, l("URL"), extractURL(service))
+	if service.Status.Address != nil {
+		url := service.Status.Address.GetURL()
+		dw.WriteColsLn(printers.LEVEL_0, l("Address"), url.String())
+	}
+	writeMapDesc(dw, printers.LEVEL_0, service.Labels, l("Labels"), "")
+	writeMapDesc(dw, printers.LEVEL_0, service.Annotations, l("Annotations"), "")
```

Comment:
One of thing I observed while testing this PR is, the annotations are configured at two places
`.metadata.annotations` (current)
and
`.spec.metadata.annotations`

the later one is where we set the autoscaling values  (`autoscaling.knative.dev/target: "10"`)
I think we should print these values as well part of `show`

Reply:
Yes, good idea. We should however show it in a form as they are provided during service creation/update, i.e. 'typed'

Reply:
or altogether a section in show's output which shows up only if autoscaling is configured for the service? (same for min-scale/max-scale)

Reply:
implemented as navid suggests